### PR TITLE
SPI RW registers callback

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: libraries/AP_HAL/Device.h [hunk 1]
```diff
@@ -141,8 +146,13 @@ class AP_HAL::Device {
      */
     bool read_registers(uint8_t first_reg, uint8_t *recv, uint32_t recv_len)
     {
+        uint8_t read_reg = first_reg;
         first_reg |= _read_flag;
-        return transfer(&first_reg, 1, recv, recv_len);
+        bool result = transfer(&first_reg, 1, recv, recv_len);
```
Comment (tridge) [inline libraries/AP_HAL/Device.h:151]: can you see if moving this function to the cpp file makes the flash impact lower?

File: libraries/AP_HAL/Device.h [hunk 2]
```diff
@@ -157,7 +167,19 @@ class AP_HAL::Device {
         if (checked) {
             set_checked_register(reg, val);
         }
-        return transfer(buf, sizeof(buf), nullptr, 0);
+        bool result = transfer(buf, sizeof(buf), nullptr, 0);
```
Comment (tridge) [inline libraries/AP_HAL/Device.h:170]: again, in cpp if possible

**Review by tridge — CHANGES_REQUESTED** [review]
I like this a lot more than the previous approach

File: libraries/AP_HAL/Device.h (hunk 1 above)
Comment (bugobliterator) [reply libraries/AP_HAL/Device.h]: This saved quite a decent amount of flash, so I have done so for all the methods in Device.h, basically moved all the methods that decrease flash cost.

**Review by tridge — APPROVED** [review]

**Pull request merged**